src/transport/raw/tests/TestMessageHeader.cpp

```diff
@@ -50,7 +50,7 @@ void TestPayloadHeaderInitialState(nlTestSuite * inSuite, void * inContext)
 
     NL_TEST_ASSERT(inSuite, header.GetMessageType() == 0);
     NL_TEST_ASSERT(inSuite, header.GetExchangeID() == 0);
-    NL_TEST_ASSERT(inSuite, header.GetProtocolID() == 0);
+    NL_TEST_ASSERT(inSuite, header.GetProtocolID() == static_cast<uint16_t>(Protocols::kProtocol_NotSpecified));
```

Comment:
This cast doesn't seem valid: `kProtocol_NotSpecified` is not in fact a value that fits in 16 bits....

Reply:
Seems like this is a bug in the PayloadHeader API. It stores ProtocolID as a 16 bit value when it should be 32 according to spec. That seems like it could be out of scope of this PR though. I could file an issue and follow up PR instead. what do you think?

Reply:
PayloadGeader stores an optional 16-bit vendor id and a 16-bit protocol id.  This is also what the spec defines for the payload header...  How this interacts with the concept of "protocol id" in the rest of the code, which is a combination of those two values, is a good question, and I agree that it's out of scope for this PR.  

We do have a bunch of code that tests the protocol id without considering the vendor id.  That code is probably all broken if someone sends a vendor-scoped protocol id.  Again, seems out of scope for this PR.

For this PR, maybe just stick to using 0xFFFF for the default value for now and file followups to make this stuff more consistent across the spec and less footgunny?

Reply:
Thanks, I was reading the PayloadHeader too fast and missed that it splits VendorID out.

For this code and the initializer, I can mask out the lower 16 bits explicitly.